### PR TITLE
Handle duplicate column names in MSSQL and SQLite query results.

### PR DESCRIPTION
**Pull request by corivera**
Fixes https://github.com/dotnet/interactive/issues/983

Table Schemas require column names to be unique, but SQL query results can have duplicate columns names due to aliasing or being unnamed. This change solves that problem by appending a number to the end of a duplicate column name, similar to what would happen if you download duplicate files in a web browser.

### CODE REVIEW

File: src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/SqlKernelsExtensionTests.cs [hunk 1]
```diff
@@ -90,6 +90,17 @@ private static IEnumerable<IEnumerable<IEnumerable<(string name, object value)>>
...
 
+        [Fact]
+        public void can_alias_duplicate_column_names_in_sql_results()
+        {
+            var testNames = new string[] { "apple", "banana", "apple", "orange", "orange", "BANANA", "orange", "BaNaNa" };
+            var expectedAliases = new string[] { "apple", "banana", "apple (1)", "orange", "orange (1)", "BANANA", "orange (2)", "BaNaNa" };
+
+            SqlKernelUtils.AliasDuplicateColumnNames(testNames);
+
+            Assert.Equal(expectedAliases, testNames);
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/SqlKernelsExtensionTests.cs:101]: Please use FluentAssertions, e.g. `expectedAliases.Should().BeEquivalentTo(testNames);`

File: src/Microsoft.DotNet.Interactive.ExtensionLab/SqlKernelUtils.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.DotNet.Interactive.ExtensionLab
+{
+    public static class SqlKernelUtils
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.ExtensionLab/SqlKernelUtils.cs:8]: This should be kept `internal`.

File: src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/SqlKernelsExtensionTests.cs [hunk 3]
```diff
@@ -90,6 +90,17 @@ private static IEnumerable<IEnumerable<IEnumerable<(string name, object value)>>
             } while (reader.NextResult());
         }
 
+        [Fact]
+        public void can_alias_duplicate_column_names_in_sql_results()
+        {
+            var testNames = new string[] { "apple", "banana", "apple", "orange", "orange", "BANANA", "orange", "BaNaNa" };
+            var expectedAliases = new string[] { "apple", "banana", "apple (1)", "orange", "orange (1)", "BANANA", "orange (2)", "BaNaNa" };
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/SqlKernelsExtensionTests.cs:97]: It seems like we should do one of the following:

* Also number the first occurrence of "apple", or
* Start the numbering at 2.

Comment (jonsequitur) [reply src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/SqlKernelsExtensionTests.cs:97]: By testing this method in isolation, we're not actually guaranteeing that the end-to-end works. A more robust approach would be to verify against the actual serialized output.

File: src/Microsoft.DotNet.Interactive.ExtensionLab/SqlKernelUtils.cs (hunk 2 above)
Comment (corivera) [reply src/Microsoft.DotNet.Interactive.ExtensionLab/SqlKernelUtils.cs:8]: It wouldn't be visible to the test project then. Is that alright with you?

File: src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/SqlKernelsExtensionTests.cs (hunk 3 above)
Comment (corivera) [reply src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/SqlKernelsExtensionTests.cs:97]: I'll do the second option. Doing the first would require two passes through the names array.

File: src/Microsoft.DotNet.Interactive.ExtensionLab/SqlKernelUtils.cs (hunk 2 above)
Comment (jonsequitur) [reply src/Microsoft.DotNet.Interactive.ExtensionLab/SqlKernelUtils.cs:8]: That's ideal. It's an implementation detail and shouldn't be tested directly.

File: src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/SqlKernelsExtensionTests.cs (hunk 3 above)
Comment (jonsequitur) [reply src/Microsoft.DotNet.Interactive.ExtensionLab.Tests/SqlKernelsExtensionTests.cs:97]: If one approach is clearer or more common than the other, I wouldn't sweat an extra loop through the array. The perf hit there is negligible especially on top of network I/O and serialization costs.

**Review by jonsequitur — APPROVED** [review]

**Pull request merged**